vllm_detector_adapter/generative_detectors/granite_guardian.py

```diff
@@ -188,6 +196,36 @@ def preprocess_request(  # noqa: F811
         """Granite guardian chat request preprocess is just detector parameter updates"""
         return self.__preprocess(request)
 
+    async def post_process_completion_results(
+        self, response: ChatCompletionResponse, scores: List[float], detection_type: str
+    ) -> Tuple[ChatCompletionResponse, List[float], str, Optional[List[Dict]]]:
+        """Process Granite Guardian chat completion tags for metadata. Metadata corresponds to
+        one Dict per choice in the chat completion response. This implementation
+        does not require async, but the base impl is async
+        """
+        metadata_list = []
+        for i, choice in enumerate(response.choices):
+            content = choice.message.content
+            metadata = (
+                {}
+            )  # Avoid messing up metadata order in case content is not present
+            if content and isinstance(content, str):
+                regex_str = (
+                    f"<{self.METADATA_ATTRIBUTE}> (.*?) </{self.METADATA_ATTRIBUTE}>"
+                )
+                # Some (older) Granite Guardian versions may not contain extra information
+                # for metadata. Make sure this does not break anything
+                if metadata_search := re.search(regex_str, content):
+                    # Update choice content as necessary, removing the metadata portion
+                    response.choices[i].message.content = re.sub(
+                        regex_str, "", content
+                    ).strip()
```

Comment:
Somewhat nit: 
Can we extract this particular logic in a `get_html_tagged_data` or `get_tagged_data` ? 
So something like:
```python
def get_tagged_data(content: str) -> str
    pass
```

(we can move this function to utils etc. and unit test separately)

That way we can reuse this in different places and might be a bit cleaner, as there can be other things that would go in postprocess in future ?

Reply:
Its probably not a big deal for GG type models, since they don't actually give out a lot of data. But in future, if the text here increases, we might want to move to using regex.compile